store/tree/actions.js

```diff
@@ -5,42 +5,36 @@
 import treeData from '~/model/tree/treeData';
 
 export default {
-    setAction: ({ commit }, payload) => {
-        commit('setState', payload);
+    setRowsCount: ({ commit }, value) => {
+        commit('setRowsCount', value);
     },
     getTreeById(
         { commit, rootState },
         { treeId, onError },
     ) {
         const sortedData = treeData.sort((a, b) => a.row - b.row);
-        commit('setState', {
-            key: 'treeData',
-            value: sortedData,
-        });
+        commit('setTree', sortedData);
         const { language: userLanguageCode } = rootState.authentication.user;
         return this.app.$axios.$get(`${userLanguageCode}/trees/${treeId}`).then(() => {
             // TODO: uncomment when we have date from API
-            // commit('setState', {
-            //     key: 'treeData',
-            //     value: response.data,
-            // });
+            // commit('setTree', response.data);
         }).catch(e => onError(e.data));
     },
-    addTreeItem: ({ commit, state }, { item }) => {
+    addTreeItem: ({ commit, state }, item) => {
         const findIndex = state.treeData.findIndex(el => el.id === item.id);
         if (findIndex >= 0) {
             commit('setTreeItem', { index: findIndex, item });
         } else {
-            commit('addTreeItem', { item });
+            commit('addTreeItem', item);
         }
     },
-    removeTreeItem: ({ commit }, { index }) => {
-        commit('removeTreeItem', { index });
+    removeTreeItem: ({ commit }, id) => {
+        commit('removeTreeItem', id);
     },
-    rebuildTree: ({ commit, state }, { id }) => {
+    rebuildTree: ({ commit, state }, id) => {
         const findIndex = state.treeData.findIndex(el => el.id === id);
         const newTree = state.treeData.reduce((accumulator, current, i) => {
-            if (i === findIndex) {
+            if (i === findIndex && current.row !== 0) {
```

Comment:
Describe condition with variable

Reply:
I think you can leave it, it's short enough to understand what is going on